src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -33,6 +37,9 @@ CARMACloudPlugin::CARMACloudPlugin(string name) :PluginClient(name) {
 		base_hb = "/carmacloud/v2xhub";
 		base_req = "/carmacloud/tcmreq";
 		method = "POST";
+		_not_ACK_TCMs = std::make_shared<multimap<string, tsm5EncodedMessage>>();
+		std::thread Broadcast_t(&CARMACloudPlugin::Broadcast_TCMs, this);
+		Broadcast_t.detach();

```

Comment:
Do we want to run "Broadcast_TCMs" as a detached thread indefinitely? Should TCM broadcast be triggered by some carma-cloud event/update or by a vehicle TCR? If so do we want to make this a call in response to some event? For example the vehicle transmits a TCR, CARMACloudPlugin sends a REST request and the response triggers the broadcast right? Maybe I am miss understanding the functionality

Reply:
If will not run infinitely as there is a check for error status